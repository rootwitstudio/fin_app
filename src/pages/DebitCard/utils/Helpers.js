export const currency = 'S$';
export const currencyFormat = (amount) => {
    // eslint-disable-next-line no-underscore-dangle
    let _amount = amount?.toString();
    return currency + _amount?.replace(/(\d)(?=(\d\d\d)+(?!\d))/g, '$1,');
};
