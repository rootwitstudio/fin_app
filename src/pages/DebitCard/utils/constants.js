export const settingsData = [
    {
        id: 0,
        name: 'Top-up account',
        desc: 'Deposit money to your account to use card',
        icon: 'upload-outline'
    },
    {
        id: 1,
        name: 'Weekly spending limit',
        desc: "You haven't set any spending limit",
        icon: 'speedometer'
    },
    {
        id: 2,
        name: 'Freeze card',
        desc: 'Your debit card is currently active',
        icon: 'card-text'
    },
    {
        id: 3,
        name: 'Get a new card',
        desc: 'Your debit card is currently active',
        icon: 'card-plus-outline'
    },
    {
        id: 4,
        name: 'Deactivated Cards',
        desc: 'Your debit card is currently active',
        icon: 'credit-card-minus'
    }
];

export const preLoadAmountData = [
    {
        id: 1,
        name: '5000',
        amount: 5000
    },
    {
        id: 2,
        name: '10000',
        amount: 10000
    },
    {
        id: 3,
        name: '20000',
        amount: 20000
    }
];
