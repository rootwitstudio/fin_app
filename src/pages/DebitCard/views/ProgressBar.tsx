import * as React from 'react';
import { FunctionComponent,useEffect, useState, useRef } from 'react';
import { Text, View, StyleSheet, Animated } from 'react-native';
import { currencyFormat } from '../utils/Helpers';


interface ProgressProps {
    limit: number
    current: number
}

const Progress : FunctionComponent<ProgressProps> = ({ limit, current }) => {
    const [percent, setPercent] = useState(0);

    useEffect(() => {
        setPercent((current / limit) * 100);
    }, [limit, current]);
    const counter = useRef(new Animated.Value(0)).current;
    const countInterval = useRef(null);
    const [count, setCount] = useState(0);

    useEffect(() => {
        countInterval.current = setInterval(() => setCount((old) => old + 5), 1000);
        return () => {
            clearInterval(countInterval);
        };
    }, []);

    useEffect(() => {
        load(count);
        if (count >= percent) {
            setCount(percent);
            clearInterval(countInterval);
        }
    }, [count]);

    const load = (count) => {
        Animated.timing(counter, {
            toValue: count,
            duration: 500,
            useNativeDriver: false,
        }).start();
    };

    const width = counter.interpolate({
        inputRange: [0, 100],
        outputRange: ['0%', '100%'],
        extrapolate: 'clamp'
    });

    return (
        <View style={styles.container}>
            <View style={styles.limitDetails}>
                <Text>Debit card spending limit</Text>
                <View style={styles.flexRow}>
                    <View>
                        <Text style={styles.cl}>{currencyFormat(current)}</Text>
                    </View>
                    <View style={styles.verticalLine} />
                    <View>
                        <Text style={styles.tl}>{currencyFormat(limit)}</Text>
                    </View>
                </View>
            </View>
            <View style={styles.progressBar}>
                <Animated.View style={([StyleSheet.absoluteFill], { backgroundColor: '#01D167', width })} />
            </View>
        </View>
    );
};

export default React.memo(Progress);

const styles = StyleSheet.create({
    container: { marginTop: 30 },
    limitDetails: { flexDirection: 'row', justifyContent: 'space-between', paddingVertical: 5 },
    progressBar: {
        height: 20,
        flexDirection: 'row',
        width: '100%',
        backgroundColor: 'white',
        borderColor: '#01D167',
        borderWidth: 2,
        borderRadius: 20
    },
    verticalLine: {
        marginHorizontal: 10,
        height: '70%',
        width: 1,
        backgroundColor: '#909090'
    },
    flexRow: {
        flexDirection: 'row'
    },
    cl: { color: '#01D167', fontWeight: 'bold' },
    tl: { opacity: 0.5 }
});
