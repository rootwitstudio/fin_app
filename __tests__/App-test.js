/**
 * @format
 */

import {currency, currencyFormat} from "../src/pages/DebitCard/utils/Helpers";
import renderer from 'react-test-renderer';
import ProgressBar from "../src/pages/DebitCard/views/ProgressBar";


test('renders correctly', () => {
    // eslint-disable-next-line react/react-in-jsx-scope
    const tree = renderer.create(<ProgressBar />).toJSON();
    expect(tree).toMatchSnapshot();
});

describe('Helpers', () => {
    test('currencyFormat', () => {
        expect(currencyFormat(50)).toBe(`${currency}50`);
        expect(currencyFormat(5000)).toBe(`${currency}5,000`);
    });
});
