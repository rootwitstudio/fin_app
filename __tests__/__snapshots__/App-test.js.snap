// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "marginTop": 30,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 5,
      }
    }
  >
    <Text>
      Debit card spending limit
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "#01D167",
              "fontWeight": "bold",
            }
          }
        >
          S$undefined
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#909090",
            "height": "70%",
            "marginHorizontal": 10,
            "width": 1,
          }
        }
      />
      <View>
        <Text
          style={
            Object {
              "opacity": 0.5,
            }
          }
        >
          S$undefined
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "#01D167",
        "borderRadius": 20,
        "borderWidth": 2,
        "flexDirection": "row",
        "height": 20,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#01D167",
          "width": "0%",
        }
      }
    />
  </View>
</View>
`;
